multi_boot/unk_8D94B9C/Makefile

```diff
@@ -1,197 +1,353 @@
-
-MID_SRCS := $(wildcard $(MID_SUBDIR)/*.mid)
-MID_OBJS := $(patsubst $(MID_SUBDIR)/%.mid,$(MID_BUILDDIR)/%.o,$(MID_SRCS))
-
-OBJS := $(C_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
+OBJS     := $(C_OBJS) $(C_ASM_OBJS) $(ASM_OBJS) $(DATA_ASM_OBJS) $(SONG_OBJS) $(MID_OBJS)
 OBJS_REL := $(patsubst $(OBJ_DIR)/%,%,$(OBJS))
 
-TOOLDIRS := $(filter-out ../../tools/agbcc ../../tools/binutils,$(wildcard ../../tools/*))
-TOOLBASE = $(TOOLDIRS:../../tools/%=%)
-TOOLS = $(foreach tool,$(TOOLBASE),../../tools/$(tool)/$(tool)$(EXE))
-
-.PHONY: all rom tools clean-tools mostlyclean clean compare tidy $(TOOLDIRS)
+SUBDIRS  := $(sort $(dir $(OBJS)))
+$(shell mkdir -p $(SUBDIRS))
 
-MAKEFLAGS += --no-print-directory
-
-AUTO_GEN_TARGETS :=
-
-all: tools rom
+# Pretend rules that are actually flags defer to `make all`
+modern: all
+compare: all
 
+# Other rules
 rom: $(ROM)
 ifeq ($(COMPARE),1)
-	@$(SHA1) $(BUILD_NAME).sha1
+	@$(SHA1) $(FILE_NAME).sha1
 endif
 
-tools: $(TOOLDIRS)
+syms: $(SYM)
 
-$(TOOLDIRS):
-	@$(MAKE) -C $@
+clean: tidy clean-tools clean-generated clean-assets
 
-# For contributors to make sure a change didn't affect the contents of the ROM.
-compare: all
+clean-assets:
+	rm -f $(MID_SUBDIR)/*.s
+	rm -f $(DATA_ASM_SUBDIR)/layouts/layouts.inc $(DATA_ASM_SUBDIR)/layouts/layouts_table.inc
+	rm -f $(DATA_ASM_SUBDIR)/maps/connections.inc $(DATA_ASM_SUBDIR)/maps/events.inc $(DATA_ASM_SUBDIR)/maps/groups.inc $(DATA_ASM_SUBDIR)/maps/headers.inc
+	find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
 
-mostlyclean: tidy
-	@rm -f sound/direct_sound_samples/*.bin
-	@rm -f $(SONG_OBJS) $(MID_SUBDIR)/*.s
-	@find . \( -iname '*.1bpp' -o -iname '*.4bpp' -o -iname '*.8bpp' -o -iname '*.gbapal' -o -iname '*.lz' -o -iname '*.rl' -o -iname '*.latfont' -o -iname '*.hwjpnfont' -o -iname '*.fwjpnfont' \) -exec rm {} +
-	@rm -f $(AUTO_GEN_TARGETS)
+tidy: tidynonmodern tidymodern
 
-clean-tools:
-	@$(foreach tooldir,$(TOOLDIRS),$(MAKE) clean -C $(tooldir);)
+tidynonmodern:
+	rm -f $(ROM_NAME) $(ELF_NAME) $(MAP_NAME)
+	rm -rf $(OBJ_DIR_NAME)
 
-clean: mostlyclean clean-tools
+tidymodern:
+	rm -f $(MODERN_ROM_NAME) $(MODERN_ELF_NAME) $(MODERN_MAP_NAME)
+	rm -rf $(MODERN_OBJ_DIR_NAME)
 
-tidy:
-	@rm -f $(ROM) $(ELF) $(MAP)
-	@rm -r build/*
+# Other rules
 
-include graphics_file_rules.mk
-include songs.mk
+# NOTE: Tools must have been built prior (FIXME)
+# so you can't really call this rule directly
+generated: $(AUTO_GEN_TARGETS)
+	@: # Silence the "Nothing to be done for `generated'" message, which some people were confusing for an error.
 
-ifeq (,$(filter-out all rom compare,$(MAKECMDGOALS)))
--include tools
-endif
-
-baserom.gba: ../../baserom.gba
-	@dd if=$< of=$@ skip=14240668 count=125548 iflag=skip_bytes,count_bytes status=none
```

Comment:
It's impossible to build if you simply remove this. The `baserom.gba`s in subdirectories are generated based on the one in the root directory.

Reply:
Done for all the sub-makefiles. I also added a rm -f for baserom.gba in those folders.